be/src/vec/exprs/lambda_function/varray_map_function.cpp

```diff
@@ -113,75 +151,172 @@ class ArrayMapFunction : public LambdaFunction {
                     return Status::InvalidArgument(
-                            "in array map function, the input column size "
-                            "are "
-                            "not equal completely, nested column data rows 1st size is {}, {}th "
-                            "size is {}.",
-                            nested_array_column_rows, i + 1, col_array.get_data_ptr()->size());
+                        "in array map function, the input column size "
+                        "are "
+                        "not equal completely, nested column data rows 1st size is {}, {}th "
+                        "size is {}.",
+                        nested_array_column_rows, i + 1, col_array.get_data_ptr()->size());
                 }
             }
-
-            // insert the data column to the new block
-            ColumnWithTypeAndName data_column {col_array.get_data_ptr(), col_type.get_nested_type(),
-                                               "R" + array_column_type_name.name};
-            lambda_block.insert(std::move(data_column));
+            lambda_datas[i] = col_array.get_data_ptr();
+            names.push_back("R" + array_column_type_name.name);
+            data_types.push_back(col_type.get_nested_type());
         }
 
-        //3. child[0]->execute(new_block)
-        RETURN_IF_ERROR(children[0]->execute(context, &lambda_block, result_column_id));
+        ColumnPtr result_col = nullptr;
+        DataTypePtr res_type;
+        std::string res_name;
+
+        //process first row
+        args.array_start = (*args.offsets_ptr)[args.current_row_idx - 1];
+        args.cur_size = (*args.offsets_ptr)[args.current_row_idx] - args.array_start;
+
+        while (args.current_row_idx < block->rows()) {
+            Block lambda_block;
+            for (int i = 0; i < names.size(); i++) {
+                ColumnWithTypeAndName data_column {data_types[i], names[i]};
+                lambda_block.insert(std::move(data_column));
+            }
+
+            MutableBlock m_lambda_block(&lambda_block);
+            MutableColumns& columns = m_lambda_block.mutable_columns();
+            while (columns[gap]->size() < batch_size) {
+                long max_step = batch_size - columns[gap]->size();
+                long current_step = std::min(max_step, (long)(args.cur_size - args.current_offset_in_array));
+                size_t pos = args.array_start + args.current_offset_in_array;
+                for (int i = 0; i < arguments.size(); ++i) {
+                    columns[gap + i]->insert_range_from(*lambda_datas[i], pos, current_step);
+                }
+                args.current_offset_in_array += current_step;
+                args.current_repeat_times += current_step;
+                if (args.current_offset_in_array >= args.cur_size) {
+                    args.eos = true;
+                } else {
+                    _extend_data(columns, block, args, gap);
```

Comment:
the _extend_data seems call by args.eos = true && args.eos = false
why not move it outside